src/platform_impl/windows/event_loop.rs

```diff
@@ -360,6 +360,17 @@ impl<T: 'static> EventLoop<T> {
 
         let runner = &self.window_target.p.runner_shared;
 
+        // We aim to be consistent with the MacOS backend which has a RunLoop
+        // observer that will dispatch AboutToWait when about to wait for
+        // events, and NewEvents after the RunLoop wakes up.
+        //
+        // We emulate similar behaviour by treating `GetMessage` as our wait
+        // point and wake up point (when it returns) and we drain all other
+        // pending messages via `PeekMessage` until we come back to "wait" via
+        // `GetMessage`
+        //
+        runner.prepare_wait();
+
```

Comment:
I don't quite like the way it's structured in this function. Could you instead move the `timeout` calculation to the place it's actually being used? So basically revert your code and move the `let control_flow_timeout` part.

Reply:
done.  Also moved the computation of start nearer to where it is used. This is not strictly equivalent as time can pass while prepare_wait is being called. But i think this is fine.